packages/perseus/src/renderer-util.ts

```diff
@@ -33,31 +41,31 @@ export function getUpgradedWidgetOptions(
     });
 }
 
+/**
+ * Checks the given user input to see if any answerable widgets have not been
+ * "filled in" (ie. if they're empty). Another way to think about this
+ * function is that its a check to see if we can score the provided input.
+ */
 export function emptyWidgetsFunctional(
-    widgets: PerseusWidgetsMap,
+    widgets: ValidationDataMap,
     // This is a port of old code, I'm not sure why
     // we need widgetIds vs the keys of the widgets object
     widgetIds: Array<string>,
     userInputMap: UserInputMap,
     strings: PerseusStrings,
     locale: string,
 ): ReadonlyArray<string> {
-    const upgradedWidgets = getUpgradedWidgetOptions(widgets);
```

Comment:
Question: What makes this line no longer necessary? I looked through the PR, but not sure what's related

Reply:
This is not needed because this empty widgets call can (should?) only ever be called in a situation where the widget options have already been upgraded.  When we are on the client, this data will be widget options from the Renderer which have already been upgraded. When we are scoring on the server, it will be scoring data that is already upgraded. Basically, I don't think this function should be exported outside of Perseus.